functions/src/index.ts

```diff
@@ -166,3 +177,136 @@ export const loadFirestore = functions.https.onRequest(
     response.status(200).send();
   }
 );
+
+/*
+ * Listen for changes to the User document
+ * and initiate the appropriate Glean ping(s)
+ */
+export async function handleUserChangesImpl(
+  change: Change<DocumentSnapshot>,
+  context: EventContext
+): Promise<boolean> {
+  const userID = context.params.userID;
+  const rallyID = await getRallyIdForUser(userID);
+  if (!rallyID) {
+    // Without Rally ID, we can't make any Glean pings
+    // This is bad and should be flagged for inspection
+    throw new Error(
+      `Unable to obtain Rally ID for user ID ${userID}. Aborting Glean ping process.`
+    );
+  }
+
+  // Get an object with the current document value.
+  // If the document does not exist, it has been deleted.
+  const newUser = change.after.exists ? change.after.data() : null;
+
+  // Get the old document, to compare the enrollment state.
+  const oldUser = change.before.exists ? change.before.data() : null;
+
+  if (!newUser || (oldUser && oldUser.enrolled === true && !newUser.enrolled)) {
+    // User document has been deleted
+    functions.logger.info(`Sending unenrollment ping for user ID ${userID}`);
+    await gleanPings.platformUnenrollment(rallyID);
+    return true;
+  }
+
+  if ((!oldUser || !oldUser.enrolled) && newUser.enrolled === true) {
+    // User just enrolled
+    functions.logger.info(`Sending enrollment ping for user ID ${userID}`);
+    await gleanPings.platformEnrollment(rallyID);
+  }
+
+  if (
+    !isDeepStrictEqual(
+      oldUser && oldUser.demographicsData,
+      newUser && newUser.demographicsData
+    )
+  ) {
+    // User updated demographicsData
+    functions.logger.info(`Sending demographics ping for user ID ${userID}`);
+    await gleanPings.demographics(rallyID, newUser && newUser.demographicsData);
+  }
+
+  return true;
+}
+
+export const handleUserChanges = functions.firestore
+  .document("users/{userID}")
+  .onWrite(handleUserChangesImpl);
+
+/*
+ * Listen for changes to the Study document
+ * and initiate the appropriate Glean ping(s)
+ */
+export async function handleUserStudyChangesImpl(
+  change: Change<DocumentSnapshot>,
+  context: EventContext
+): Promise<boolean> {
+  const userID = context.params.userID;
+  const firebaseStudyID = context.params.studyID;
+  const rallyID = await getRallyIdForUser(userID);
+  if (!rallyID) {
+    // Without Rally ID, we can't make any Glean pings
+    // This is bad and should be flagged for inspection
+    throw new Error(
+      `Unable to obtain Rally ID for user ID ${userID}. Aborting Glean ping process.`
+    );
+  }
+
+  // Get an object with the current document value.
+  // If the document does not exist, it has been deleted.
+  const newStudy = change.after.exists ? change.after.data() : null;
+
+  // Get the old document, to compare the enrollment state.
+  const oldStudy = change.before.exists ? change.before.data() : null;
+
+  const studyID =
```

Comment:
You can do this:
const studyId = (newStudy && newStudy.studyId) || (oldStudy && oldStudy.studyId)